src/main/java/org/folio/helper/PurchaseOrderLineHelper.java

```diff
@@ -736,13 +737,19 @@ private Future<Void> validateUserUnaffiliatedLocationUpdates(CompositePoLine upd
     return getUserTenantsIfNeeded(requestContext)
       .compose(userTenants -> {
         if (CollectionUtils.isEmpty(userTenants)) {
+          logger.info("validateUserUnaffiliatedLocationUpdates:: User tenants is empty");
           return Future.succeededFuture();
         }
         var storageUnaffiliatedLocations = extractUnaffiliatedLocations(storedPoLine.getLocations(), userTenants);
         var updatedUnaffiliatedLocations = extractUnaffiliatedLocations(updatedPoLine.getLocations(), userTenants);
+        logger.info("validateUserUnaffiliatedLocationUpdates:: Found unaffiliated location tenant ids: stored: '{}', updated: '{}'",
+          storageUnaffiliatedLocations.stream().map(Location::getTenantId).toList(),
+          updatedUnaffiliatedLocations.stream().map(Location::getTenantId).toList());
         if (!SetUtils.isEqualSet(storageUnaffiliatedLocations, updatedUnaffiliatedLocations)) {
+          logger.info("validateUserUnaffiliatedLocationUpdates:: User is not affiliated with all locations on the POL");
```

Comment:
better to add some identifier because for some cases it can be problematic to find correct logs. Propose to add updatedPoLine.getId().
In this case for example QA just can get us po line id, and we can use it in search to find all information what is going on with that po line id.